covalent/cloud_resource_manager/core.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2023 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+
+import importlib
+import os
+import shutil
+import subprocess
+from configparser import ConfigParser
+from pathlib import Path
+from types import ModuleType
+from typing import Any, Dict, Optional
+
+from covalent._shared_files.config import set_config
+from covalent._shared_files.exceptions import CommandNotFoundError
+from covalent.executor import _executor_manager
+
+
+def get_executor_module(executor_name: str) -> ModuleType:
+    """
+    Get the executor module from the executor name
+
+    Args:
+        executor_name: Name of the executor
+
+    Returns:
+        The executor module
+
+    """
+    return importlib.import_module(
+        _executor_manager.executor_plugins_map[executor_name].__module__
+    )
+
+
+def get_converted_value(value: str) -> Any:
```

Comment:
@kessler-frost is there any way we can leverage the pydantic models to do this conversion?

Reply:
Done.